### PR TITLE
[i18n-zh] Sync source/zh/index.md

### PR DESCRIPTION
**Pull request by windsonsea**
Sync with the [recent en history](https://github.com/huggingface/transformers/commits/main/docs/source/en/index.md).

```
docs/source/zh/index.md
```

For references to model_doc , use the en url currently. It shall be updated after translation in the future.

cc @stevhliu @Fan-Lin @jiaqiw09

### CODE REVIEW

File: docs/source/zh/index.md [hunk 1]
```diff
@@ -37,362 +37,285 @@ rendered properly in your Markdown viewer.
 
 ## 目录
 
-这篇文档被组织为以下5个章节:
+这篇文档被组织为以下 5 个章节:
```
Comment (Fan-Lin) [inline docs/source/zh/index.md:40]:
```suggestion
这篇文档由以下 5 个章节组成:
```

Comment (windsonsea) [reply docs/source/zh/index.md]: Thanks, updated

**Review by stevhliu — APPROVED** [review]
Nice job, thanks!

**Pull request merged**